frontend/packages/console-plugin-sdk/src/codegen/active-plugins.ts

```diff
@@ -13,8 +15,56 @@ import { Extension, ActivePlugin } from '../typings';
 import { trimStartMultiLine } from '../utils/string';
 import { consolePkgScope, PluginPackage } from './plugin-resolver';
 
+const getExtensionsFilePath = (pkg: PluginPackage) => path.resolve(pkg._path, extensionsFile);
+
+/**
+ * Guess the file path of the module (e.g., the extension,
+ * any barrel/index file) based on the given base path.
+ *
+ * Returns the base path if no file is found.
+ */
+const guessModuleFilePath = (basePath: string) => {
+  const extensions = ['.tsx', '.ts', '.jsx', '.js'];
+
+  // Sometimes the module is an index file, but the exposed module path only specifies the directory.
+  // In that case, we need an explicit check for the index file.
+  const indexModulePaths = ['index.ts', 'index.js'].map((i) => path.resolve(basePath, i));
+
+  const pathsToCheck = [...indexModulePaths, ...extensions.map((ext) => `${basePath}${ext}`)];
+
+  for (const p of pathsToCheck) {
+    if (fs.existsSync(p)) {
+      return p;
+    }
+  }
+
+  // A file couldn't be found, return the original module path. A compilation warning
+  // may be pushed by `getActivePluginsModuleData`
+  return basePath;
```

Comment:
You can init the `diagnostics` object early in `getActivePluginsModuleData` function and pass it down to this function, so we can `warnings.push()` a message here saying we were not able to find a file for the given module path.

This warning can be useful to detect problems in exposed modules of Console static plugins.